scripts/obtain/dotnet-install.ps1

```diff
@@ -281,7 +292,13 @@ function Get-List-Of-Directories-And-Versions-To-Unpack-From-Dotnet-Package([Sys
 function Extract-Dotnet-Package([string]$ZipPath, [string]$OutPath) {
     Say-Invocation $MyInvocation
 
-    Add-Type -Assembly System.IO.Compression.FileSystem | Out-Null
+    try {
+        Add-Type -Assembly System.IO.Compression.FileSystem | Out-Null
+    }
+    catch {
+        # Loading FileSystem is unnecessary and will fail on Nano Server
+    }
+
     Set-Variable -Name Zip
     try {
         $Zip = [System.IO.Compression.ZipFile]::OpenRead($ZipPath)
```

Comment:
I am not an expert on this but my understanding is that Nano Server has a different flavor of Powershell - Core Edition built on .NET Core - vs the Desktop Edition build on .NET Framework.  Add-Type works on the Desktop Edition because it finds the assembly in the GAC.  On Nano, Add-Type is unable to resolve the assembly because it isn't GAC'd or does it reside in the active directory therefore an exception is thrown.  Core Edition is however able to resolve type references from the base libraries without having to explicitly load the assemblies.  They are automatically resolved from assemblies in C:\Windows\system32\DotNetCore\v1.0.